Src/FluentAssertions/Equivalency/MultiDimensionalArrayEquivalencyStep.cs

```diff
@@ -13,7 +13,7 @@ internal class MultiDimensionalArrayEquivalencyStep : IEquivalencyStep
 {
     public EquivalencyResult Handle(Comparands comparands, IEquivalencyValidationContext context, IEquivalencyValidator nestedValidator)
     {
-        if (comparands.Expectation is not Array expectationAsArray || expectationAsArray?.Rank is 1)
+        if (comparands.Expectation is not Array { Rank: not 1 } expectationAsArray)
```

Comment:
Although Rider suggested the same fix, I really find it very hard to read. It's almost like a double negative.

Reply:
I agree, went overboard with this.
reverted and only removed the redundant null check